examples/platforms/nrf528xx/src/radio.c

```diff
@@ -452,19 +452,35 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
 otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
 {
-    bool result = true;
+    bool    result = true;
+    otError error  = OT_ERROR_NONE;
 
     aFrame->mPsdu[-1] = aFrame->mLength;
 
     nrf_802154_channel_set(aFrame->mChannel);
 
-    if (aFrame->mInfo.mTxInfo.mCsmaCaEnabled)
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+    if ((otPlatRadioGetCaps(NULL) & OT_RADIO_CAPS_TRANSMIT_TIMING) != 0 && aFrame->mInfo.mTxInfo.mTxDelay != 0)
     {
-        nrf_802154_transmit_csma_ca_raw(&aFrame->mPsdu[-1]);
+        if (!nrf_802154_transmit_raw_at(&aFrame->mPsdu[-1], aFrame->mInfo.mTxInfo.mCsmaCaEnabled,
+                                        aFrame->mInfo.mTxInfo.mTxDelayBaseTime, aFrame->mInfo.mTxInfo.mTxDelay,
+                                        aFrame->mChannel))
+        {
+            error = OT_ERROR_INVALID_STATE;
+        }
+        otEXPECT(false);
```

Comment:
Why jump to exit here? In this way, a CSL transmission will not call `otPlatRadioTxStarted()` callback.